src/app/librarian/librarian.component.ts

```diff
@@ -1,15 +1,203 @@
 export class LibrarianComponent implements OnInit {
+  branches: any;
+  branchId: number;
+  bookId: number;
+  noOfCopies: number;
+  branchName: string;
+  branchAddress: string;
+  selectedBranch: any;
+  bookCopies: any;
+  books: any;
+  totalBooks: any;
+
+  private modalRef: NgbModalRef;
+  closeResult: any;
+  searchString: any;
+  updateBranchForm: FormGroup;
+  addBookCopiesForm: FormGroup;
+  searchBranchForm: FormGroup;
+  dropdownSettings: any;
+  errMsg: any;
+  constructor(private lmsService: LmsService,
+    private modalService: NgbModal,
+    private fb: FormBuilder
+  ) {
+    this.dropdownSettings = {
+      singleSelection: true,
+      idField: "bookId",
+      textField: "title",
+      itemsShowLimit: 5,
+      allowSearchFilter: true,
+    }
+  }
 
-  constructor() { }
 
   ngOnInit() {
+    this.loadAllBranches();
+    this.loadAllBooks();
+    this.initializeFormGroup();
+  }
+
+  loadAllBranches() {
+    this.lmsService
+      .get(`${environment.adminBackendUrl}${environment.readBanchUri}`)
+      .subscribe(
+        (res) => {
+          this.branches = res;
+          this.branches.forEach(branch => {
+            branch.books = [];
+            branch.bookCopies.forEach(element => {
+              branch.books.push({
+                title: element.title,
+                noOfCopies: element.noOfCopies,
+              });
+            });
+          });
+        },
+        (error) => {
+          debugger;
+        }
+      );
+  }
+
+  loadAllBooks() {
+    this.lmsService
+      .get(`${environment.adminBackendUrl}${environment.readBookUri}`)
+      .subscribe(
+        (res) => {
+          this.totalBooks = res;
+
+        },
+        (error) => {
+          debugger;
+        }
+      );
+  }
+
+  initializeFormGroup() {
+    this.updateBranchForm = new FormGroup({
+      branchName: new FormControl(this.branchName, [
+        Validators.required,
+        Validators.maxLength(45),
+        Validators.minLength(3),
+      ]),
+      branchAddress: new FormControl(this.branchAddress, [
+        Validators.required,
+        Validators.maxLength(45),
+        Validators.minLength(3),
+      ]),
+      bookId: new FormControl(this.bookId),
+      noOfCopies: new FormControl(this.noOfCopies),
+      branchId: new FormControl(this.branchId),
+      books: new FormControl(this.books),
+    });
   }
 
+
+  updateBranch() {
+    const branch = {
+      branchId: this.updateBranchForm.value.branchId,
+      branchName: this.updateBranchForm.value.branchName,
+      branchAddress: this.updateBranchForm.value.branchAddress,
+    }
+    this.lmsService
+      .post(`${environment.libUrl}${environment.updateBanchUri}`, branch)
+      .subscribe(
+        (res) => {
+          console.log(res);
+        },
+        (error) => {
+          console.log(error);
+        }
+      );
+  }
+
+
+  openUpdate(content, obj) {
+    if (obj !== null) {
```

Comment:
This check isn't needed for this component. obj will never be null since a branch is always passed when this function is called.